controllers/volumereplication_controller.go

```diff
@@ -317,6 +328,48 @@ func (r *VolumeReplicationReconciler) SetupWithManager(mgr ctrl.Manager, cfg *co
 		WithEventFilter(pred).Complete(r)
 }
 
+func (r *VolumeReplicationReconciler) waitForCrds() error {
+	logger := r.Log.WithName("checkingDependencies")
+
+	err := r.waitForVolumeReplicationResource(logger, volumeReplicationClass)
+	if err != nil {
+		logger.Error(err, "failed to wait for VolumeReplicationClass CRD")
+		return err
+	}
+
+	err = r.waitForVolumeReplicationResource(logger, volumeReplication)
+	if err != nil {
+		logger.Error(err, "failed to wait for VolumeReplication CRD")
+		return err
+	}
+	return nil
+}
+
+func (r *VolumeReplicationReconciler) waitForVolumeReplicationResource(logger logr.Logger, resourceName string) error {
+	isResourceExists := false
+	unstructuredResource := &unstructured.UnstructuredList{}
+	unstructuredResource.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   replicationv1alpha1.GroupVersion.Group,
+		Kind:    resourceName,
+		Version: replicationv1alpha1.GroupVersion.Version,
+	})
+	for !isResourceExists {
+		err := r.Client.List(context.TODO(), unstructuredResource)
+		if err != nil {
+			if meta.IsNoMatchError(err){
+				logger.Info("resource does not exist", "Resource",  resourceName)
+				time.Sleep(5 * time.Second)
+			} else {
```

Comment:
We could remove the both the else statement in line 362 and 366 to reduce the nesting. 

```
 if err != nil {
      if meta.IsNoMatchError(err){
	  logger.Info("resource does not exist", "Resource",  resourceName)
	  time.Sleep(5 * time.Second)
          return nil
        } 
     logger.Error(err, "got an unexpected error while waiting for resource", "Resource", resourceName)
     return err
  } 
   isResourceExists = true
```

Reply:
if the error is noMatch we need to continue not to return nil. Maybe we can enhance that in the next PR. I can send one.

Reply:
@sp98 we need them since we don't want to stop when the resource does not exists and only exit when there is a failure that doesn't relate to resource existence or that if the resource exists, otherwise we will want to stay in the loop